tests/quantization/ggml/test_ggml.py

```diff
@@ -503,6 +508,75 @@ def test_falcon7b_weights_conversion_fp16(self):
                 self.assertTrue(original_params.shape == quantized_state_dict[layer_name].shape)
                 torch.testing.assert_close(original_params, quantized_state_dict[layer_name])
 
+    def test_stablelm_q4_k_m(self):
+        model = AutoModelForCausalLM.from_pretrained(
+            self.stablelm_model_id,
+            gguf_file=self.q4_k_m_stablelm_model_id,
+            device_map="auto",
+            torch_dtype=torch.float16,
+        )
+
+        tokenizer = AutoTokenizer.from_pretrained(self.stablelm_model_id, gguf_file=self.q4_k_m_stablelm_model_id)
+        text = tokenizer(self.example_text, return_tensors="pt").to(torch_device)
+        out = model.generate(**text, max_new_tokens=10)
+
+        EXPECTED_TEXT = "Hello-\nI am trying to create a new user"
+        self.assertEqual(tokenizer.decode(out[0], skip_special_tokens=True), EXPECTED_TEXT)
+
+    def test_stablelm_fp16(self):
+        original_model = AutoModelForCausalLM.from_pretrained(
+            self.original_stablelm2_model_id,
+            torch_dtype=torch.float16,
+        )
+
+        converted_model = AutoModelForCausalLM.from_pretrained(
+            self.stablelm2_model_id,
+            gguf_file=self.fp16_stablelm2_model_id,
+            torch_dtype=torch.float16,
+            # for precise comparison it is required to use the original model config
+            # as quantized one is different in parameters: use_parallel_residual and use_qkv_bias
+            # and it highly influences on the output results
```

Comment:
I'm not use to understand why we need to use the original model config ? Since we are using the fp16 model, it shouldn't be quantized.

Reply:
Because even in fp16 model we use config from gguf file and it is quite different compared with original model config. For example, `use_parallel_residual = True` in quantazied one, while in original model it is false. If it is true, the models skips creating post_attention_layernorm. And `use_qkv_bias = False` in quantized one, and True in original model. If it is false, it skips using q, k, v biases. That's why we explicitly use original model config in order to have all weights and biases correctly created for fair comparison.

Reply:
Thanks for the explanation ! This makes sense to me. The issue is on the conversion script from hf to gguf then ? If one specify fp16, it shoudn't modify `use_parallel_residual` and `use_qkv_bias`.

Reply:
So, I suppose yes and no, use_parallel_residual = True is specified in conversion script by default `self.gguf_writer.add_parallel_residual(hparams["use_parallel_residual"] if "use_parallel_residual" in hparams else True)`, `use_qkv_bias` is not specified at all, so by default it is False according to config implementation.

Reply:
Oh thx ! That will be very helpful for the next models we add !